plugins/obs-nvenc/nvenc-opts-parser.c

```diff
@@ -0,0 +1,235 @@
+#include "nvenc-internal.h"
+
+#include <stdio.h>
+
+/* NVIDIA likes their bitfields and unfortunately the only way to do this is macros.
+ * So don't blame me for that.  */
+
+#define APPLY_BIT_OPT(opt_name, bits)                                         \
+	if (strcmp(opt->name, #opt_name) == 0) {                              \
```

Comment:
Why can't this be an inline function (per the comment above)? The macro is used quite often below which means that none of those steps can be debugged (except in raw assembly)?

Reply:
You can't get a pointer to a specific value in a bitfield. I'm also not really concerned about the debugability of setting a value on a struct member.